src/main/java/de/eldoria/semvertools/Identifier.java

```diff
@@ -6,17 +6,44 @@
 
 package de.eldoria.semvertools;
 
+import de.eldoria.semvertools.util.Integers;
 import org.jetbrains.annotations.ApiStatus;
 
+import java.util.Optional;
+import java.util.function.Supplier;
+
 @ApiStatus.NonExtendable
 public interface Identifier extends Comparable<Identifier> {
 
   static Identifier of(int num) {
+    Integers.assertNonNegative(num);
     return new NumericalIdentifier(num);
   }
 
   static Identifier of(String alphanumeric) {
-    return new AlphanumericIdentifier(alphanumeric);
+    return Integers.parseNonNegativeInt(alphanumeric)
```

Comment:
Why not switch to Optional<Integer> then? Seems to have a large advantage and isn't more verbose at all.